Fb sdk update

enginvardar opened the pull request:
Facebook SDK used for Facebook module was outdated and abandoned by Facebook, This PR updates updates the Facebook Module to the latest Facebook SDK and adds name option to test user to be created.

enginvardar commented:
rerun travis

samdark commented on tests/unit/Codeception/Module/FacebookTest.php:
```diff
@@ -11,12 +11,13 @@
-        'app_id'    => '460287924057084',
-        'secret'    => 'e27a5a07f9f07f52682d61dd69b716b5',
+        'app_id' => '1584417941835888',
+        'secret' => '17e014121a007dc5d149ada7f9201d07',
```
Are you sure these should be changed? I guess not.

samdark commented on tests/unit/Codeception/Module/FacebookTest.php:
```diff
@@ -11,12 +11,13 @@
             'permissions' => array(
-                'publish_stream'
-            )
+                'public_profile',
```
There's no need for any special permission for it. Public profile is there by default in v4 API.

samdark commented:
Overall looks OK except application ID change and unused permission.

enginvardar commented:
@samdark i've changed back application id and removed redundant permission, also i've updated documentation and find out that if test_user config does not exist test fail because of not found name key and fixed that too.

samdark commented:
Great. Looks good to me now.

Will be checked once again by @DavertMik when he'll be back from vacation and then merged. Thanks!

tiger-seo commented on src/Codeception/Lib/Driver/Facebook.php:
```diff
@@ -15,93 +18,98 @@ public function __construct($config, $logCallback = null)
+     * @return array
      */
-    protected function setPersistentData($key, $value)
+    public function createTestUser($name, array $permissions)
```
name isn't required by Facebook, so i think it shouldn't be required by Codeception ;)

enginvardar replied:
Actually name is optional, if it's not set, default is 'Codeception Testuser' in module _initialize, i think defining test users name may help during tests.

DavertMik replied:
If the name is optional then it should be passed as optional param as well, I think.

tiger-seo commented on src/Codeception/Lib/Driver/Facebook.php:
```diff
@@ -15,93 +18,98 @@ public function __construct($config, $logCallback = null)
+            '/me/feed',
+            $parameters
+        )->getRawResponse();
+        return json_decode($response, true);
```
one empty line should be before return;  please, fix :)

DavertMik commented:
As for me this looks OK. @tiger-seo what about you? Should we merge or there are things that should be taken into account?

tiger-seo commented:
it's ok ;)

DavertMik commented:
Thanks! Merged
